### PR TITLE
pickaxe, grenade, distant explosion, and access denied sound fixes

### LINKED ISSUE #6086: Explosions play only the distant explosion sound effect
### Environment
- Game Version: latest staging (21022100)

### Detailed description of bug

Explosions (either all or at least some) are currently playing the distant explosion sound files exclusively, no matter how close you are to the detonation.

### Steps to reproduce

1. spawn a grenade
2. let it blow up in you hand or right next to you
3. no traditional explosion sound effect, only the metal creaking and distant boom sfx.

### LINKED ISSUE #6916: Pickaxes and drills no longer make sounds when you start mining with them.
### Environment
- Game Version: Production 4119
- OS Version: Windows 10

### Detailed description of bug
Pickaxes and drills no longer make sounds when you start mining with them. Notably, their prefabs still path to the appropriate sounds.
image

### Steps to reproduce

1. Spawn as a role that can easily access pickaxes, such as a shaft miner.
2. Obtain a pickaxe and go to an asteroid or lavaland.
3. Mine rocks, notice that no sound plays when you start mining.

### LINKED ISSUE #7012: Clearance denied sounds on airlocks don't work as client.
### Environment
- Game Version: Production 4119
- OS Version: Windows 10

### Detailed description of bug
When you try to use doors that you don't have clearance for, there will not be a distinctive beep that accompanies flashing red lights to indicate that your clearance was denied. This only occurs for clients. Hosts still receive the sound. Possibly related to the sound portion of #6701.

### Steps to reproduce
1. Connect to a server.
2. Find a door restricted to you.
3. Try to open it. Red lights will flash but no sound will play.

### LINKED ISSUE #7018: Grenades no longer make sounds.
### Environment
- Game Version: Staging 21070919
- OS Version: Windows 10

### Detailed description of bug
Grenades currently do not produce sounds when armed nor when exploding. This gives players stationwide no feedback to when grenades are armed nor when they explode. This also applies to combustible lemons.

Note: Lack of explosion sounds may also extend to other items and objects like fuel tanks and gas canisters.

### Steps to reproduce
1. Obtain a grenade, minibomb, or combustible lemon.
2. Press Z with the explosive in question in your active hand to arm it. There will be no arming sound.
3. Wait for the explosive to explode, there will be no explosion sound.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
- [x] fixes #7012
- [x] fixes #7018
- [X] fixes #6916 
- [X] fixes #6086

CL: [Fix] fixed missing grenade sounds.
CL: [Fix] fixed clearance denied sound on airlocks.
CL: [Fix] fixed pickaxes and drills not making sounds.
CL: [Fix] fixed explosions only playing distant explosion sfx.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs [hunk 1]
```diff
@@ -206,7 +206,7 @@ private void ClientPlaySound(AddressableAudioSource sound)
 		{
 			if(CustomNetworkManager.IsHeadless) return;
 
-			_ = SoundManager.PlayAtPosition(sound, gameObject.WorldPosClient());
+			_ = SoundManager.PlayNetworkedAtPosAsync(sound, gameObject.WorldPosClient());
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: this shouldnt be networked, see line 207

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: but it needs to be networked in order for the sounds to be played on clients?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: that function is called by clients, or should be

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: will what i just pushed fix it?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: actually wait, why am i using commits as a way to ask
would just removing that and making the server play it fix it

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: scratch that, I remembered that bombs had issues with null audioparamaters so maybe this actually fixes it.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: scratch that again

**Comment by MaxIsJoe** [comment]
Converting this into a draft for a short while as more problems krept in and I'm unable to figure a fix for denied sounds until someone on discord can nudge me in the right direction or my brain works.

**Comment by MaxIsJoe** [comment]
https://user-images.githubusercontent.com/34368774/137388773-11348a5a-e840-4b7d-b112-727029b278fe.mp4

posting this here before I forget it.

Denied and warning sounds for some reason never get past the `if(CustomNetworkManager.IsHeadless) return;` on clients but for some reason other sounds do; so there's defiantly something wacky going on here.

File: UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs [hunk 2]
```diff
@@ -97,6 +98,7 @@ public void PlayAnimation(DoorUpdateType type, bool skipAnimation, bool panelExp
 			else if (type == DoorUpdateType.AccessDenied)
 			{
 				StartCoroutine(PlayDeniedAnimation());
+				Logger.Log("bitch");
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/Doors/DoorAnimatorV2.cs]: no u

**Comment by MaxIsJoe** [comment]
It's finally done.
Fully tested as well.

**Review by NoooneyDude — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**